setup.py

```diff
@@ -1,41 +1,61 @@
 #! /usr/bin/env python
 import setuptools  # noqa; we are using a setuptools namespace
-from numpy.distutils.core import setup
-import pyglmnet
+import os
+
+
+def get_version():
+    """
+    Get the version without importing, so as not to invoke dependency
+    requirements.
+    """
+    base, _ = os.path.split(os.path.realpath(__file__))
+    file = os.path.join(base, "pyglmnet", "__init__.py")
+
+    for line in open(file, "r"):
+        if "__version__" in line:
+            return line.split("=")[1].strip().strip("'").strip('"')
+
 
 descr = """Elastic-net regularized generalized linear models."""
 
-DISTNAME = 'pyglmnet'
+DISTNAME = "pyglmnet"
 DESCRIPTION = descr
-MAINTAINER = 'Pavan Ramkumar'
-MAINTAINER_EMAIL = 'pavan.ramkumar@gmail.com'
-LICENSE = 'MIT'
-DOWNLOAD_URL = 'https://github.com/glm-tools/pyglmnet.git'
-VERSION = pyglmnet.__version__
+MAINTAINER = "Pavan Ramkumar"
+MAINTAINER_EMAIL = "pavan.ramkumar@gmail.com"
+LICENSE = "MIT"
+DOWNLOAD_URL = "https://github.com/glm-tools/pyglmnet.git"
+VERSION = get_version()  # Get version without importing
+INSTALL_REQUIRES = ["numpy", "scipy", "scikit-learn"]
```

Comment:
I don't think scikit-learn is a dependency either. It's just needed for testing

Reply:
Agreed. If it's only for testing it should not be here.